### PR TITLE
Corrects MASON Rig's missing flags

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Gives the MASON RIG the proper flags that were forgotten/unknown

## Why It's Good For The Game

MASON RIG was ment to be tesla proof, Rad proof and NOT speed upable

## Changelog
:cl:
tweak: Adds missing but needed flags to MASON RIG
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
I still hate it and want it gone

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -637,12 +640,14 @@
 	armor = list("melee" = 20, "bullet" = 15, "laser" = 15, "energy" = 45, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
 	item_color = "ancient"
 	brightness_on = 16
-	clothing_flags = STOPSPRESSUREDAMAGE | THICKMATERIAL | BLOCK_GAS_SMOKE_EFFECT | ALLOWINTERNALS | SCAN_REAGENTS
 	flash_protect = 5 //We will not be flash by bombs
 	tint = 1
 	var/obj/machinery/doppler_array/integrated/bomb_radar
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	clothing_flags = STOPSPRESSUREDAMAGE | THICKMATERIAL | BLOCK_GAS_SMOKE_EFFECT | ALLOWINTERNALS | SCAN_REAGENTS | IMMUTABLE_SLOW
```
Comment (Ghommie) [inline code/modules/clothing/spacesuits/hardsuit.dm]: the mason rig helmet doesn't slowdown, why would it need the `IMMUTABLE_SLOW` flag?

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -637,12 +640,14 @@
...
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
```
Comment (Ghommie) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Same as the parent type, remove.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -627,7 +627,10 @@
 	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/storage, /obj/item/construction/rcd, /obj/item/pipe_dispenser)
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/ancient/mason
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	clothing_flags = STOPSPRESSUREDAMAGE | THICKMATERIAL | IMMUTABLE_SLOW
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
```
Comment (Ghommie) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Idem.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/modules/clothing/spacesuits/hardsuit.dm]: So people dont use a speed up on the hat

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/modules/clothing/spacesuits/hardsuit.dm]: This is to protect what ever is on its suit slot

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/modules/clothing/spacesuits/hardsuit.dm]: This is to protect the doppler array inside it

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/clothing/spacesuits/hardsuit.dm]: Ok, listen, that flag is only used to stop speed potions from being usable on certain clothings. But this hat already has a speed modifier of 0, so it won't work either ways. This flag is misleading and useless in this case, understood?

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (Ghommie) [reply code/modules/clothing/spacesuits/hardsuit.dm]: THESE ARE THE SAME RAD FLAGS THAT ALL SPACE SUITS ALREADY HAVE.

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/clothing/spacesuits/hardsuit.dm]: IDEM.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Have another free (empty) review.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**